bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -0,0 +1,443 @@
+    private @Nullable String sitename;
+    private boolean batteryConfigured = false;
+    private double batteryPower = 0;
+    private int batterySoC = 0;
+    private int batteryPrioritySoC = 0;
+    private boolean gridConfigured = false;
+    private double gridPower = 0;
+    private double homePower = 0;
+    private boolean pvConfigured = false;
+    private double pvPower = 0;
+
+    private int activePhases = 3;
+    private double chargeCurrent = 0;
+    private double chargeDuration = 0;
+    private double chargePower = 0;
+    private double chargeRemainingDuration = 0;
+    private double chargeRemainingEnergy = 0;
+    private double chargedEnergy = 0;
+    private boolean charging = false;
+    private boolean connected = false;
+    private double connectedDuration = 0;
+    private boolean enabled = false;
+    private boolean hasVehicle = false;
+    private double maxCurrent = 16;
+    private double minCurrent = 0;
+    private int minSoC = 0;
+    private String mode = "off";
+    private int phases = 3;
+    private int targetSoC = 100;
+    private @Nullable String targetTime = "null";
+    private boolean targetTimeEnabled = false;
+    private ZonedDateTime targetTimeZDT = ZonedDateTime.now();
+    private String title = "";
+    private double vehicleCapacity = 0;
+    private double vehicleOdometer = 0;
+    private boolean vehiclePresent = false;
+    private double vehicleRange = 0;
+    private int vehicleSoC = 0;
+    private String vehicleTitle = "";
+
+    public EvccHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command.equals(RefreshType.REFRESH)) {
+            refresh();
+        } else {
+            logger.debug("Handling command {} for channel {}", command, channelUID);
+            String groupId = channelUID.getGroupId();
+            if (groupId == null)
+                return;
+            String channelIdWithoutGroup = channelUID.getIdWithoutGroup();
+            int loadpoint = Integer.parseInt(groupId.toString().substring(9));
+            EvccAPI evccAPI = this.evccAPI;
+            if (evccAPI == null)
+                return;
+            switch (channelIdWithoutGroup) {
+                case CHANNEL_LOADPOINT_MODE:
+                    evccAPI.setMode(loadpoint, command.toString());
+                    break;
+                case CHANNEL_LOADPOINT_MIN_SOC:
+                    evccAPI.setMinSoC(loadpoint, Integer.parseInt(command.toString().replaceAll(" %", "")));
+                    break;
+                case CHANNEL_LOADPOINT_TARGET_SOC:
+                    evccAPI.setTargetSoC(loadpoint, Integer.parseInt(command.toString().replaceAll(" %", "")));
+                    break;
+                case CHANNEL_LOADPOINT_TARGET_TIME:
+                    if (targetTimeEnabled == true) {
+                        targetTimeZDT = new DateTimeType(command.toString()).getZonedDateTime();
```

Comment:
Hmm, nice trick. :) But if you know the format, it would be more efficient to provide the format string yourself and deal directly with `ZonedDateTime`. The `DateTimeType` string constructor use this private method for parsing:

https://github.com/openhab/openhab-core/blob/cabb3f73150d13f44d647f46af8362e396684b92/bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/DateTimeType.java#L221-L236